### PR TITLE
Track artifacts on S3 compatible storage

### PR DESCRIPTION
**Pull request by shnela**

### CODE REVIEW

File: src/neptune/new/internal/artifacts/drivers/s3.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 # See the License for the specific language governing permissions and
```
Comment (Raalsky) [inline src/neptune/new/internal/artifacts/drivers/s3.py:13]: CHANGELOG

File: src/neptune/new/internal/artifacts/drivers/s3.py [hunk 2]
```diff
@@ -36,6 +37,22 @@
...
+    def get_boto_resource():
+        """
+        User might want to use other than `AWS` `S3` providers, so we should be able to override `endpoint_url`.
+        Unfortunately `boto3` doesn't support this parameter in configuration, so we'll have to create our env variable.
+        boto3 supported config envs:
+         * https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html#using-environment-variables
+        boto3 `endpoint_url` support PR:
+         * https://github.com/boto/boto3/pull/2746
+        """
+        endpoint_url = os.getenv("S3_ENDPOINT_URL")
```
Comment (Raalsky) [inline src/neptune/new/internal/artifacts/drivers/s3.py:50]: Probably this should be a part of `neptune.new.envs`

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,7 @@
 - `get_last_run`, `get_run_url`, `get_project` and `neptune.init` marked as deprecated ([#1011](https://github.com/neptune-ai/neptune-client/pull/1011))
 - Deprecated implicit casting of objects to strings with `log` and `assign` operations ([#1028](https://github.com/neptune-ai/neptune-client/pull/1028))
 - Internally extracted legacy client to `legacy` submodule ([#1039](https://github.com/neptune-ai/neptune-client/pull/1039))
+- Track artifacts on S3 compatible storage ([#1053](https://github.com/neptune-ai/neptune-client/pull/1053))
```
Comment (Raalsky) [inline CHANGELOG.md:14]: I think this should be treated as a feature.

Comment (shnela) [reply CHANGELOG.md]: right

**Review by Raalsky — APPROVED** [review]

**Pull request merged**